std/array_list.zig

```diff
@@ -193,27 +194,105 @@ pub fn AlignedArrayList(comptime T: type, comptime A: u29) type {
             return self.pop();
         }
 
+        // an iterator which allows you to remove the current element
+        // while iterating through it.
         pub const Iterator = struct {
-            list: *const Self,
-            // how many items have we returned
-            count: usize,
+            list: *Self,
+
+            // index of the next element to return
+            next_index: usize = 0,
+
+            // the last index we removed
+            removed_index: ?usize = null,
 
             pub fn next(it: *Iterator) ?T {
-                if (it.count >= it.list.len) return null;
-                const val = it.list.at(it.count);
-                it.count += 1;
+                if (it.next_index >= it.list.len) return null;
+                const val = it.list.at(it.next_index);
+
+                it.next_index += 1;
+
```

Comment:
Should also `it.removed_index = null`?

Reply:
No need, because of `idx==it.next_index`.

Reply:
Okay; I see that avoiding `it.removed_index = null` here allows `next` to be the same as without the `remove` methods at the cost of a slightly larger iterator struct.
Fair trade off :+1: